lib/EnzymeCore/src/EnzymeCore.jl

```diff
@@ -101,6 +105,31 @@ batch_size(::BatchDuplicated{T,N}) where {T,N} = N
 batch_size(::BatchDuplicatedNoNeed{T,N}) where {T,N} = N
 Adapt.adapt_structure(to, x::BatchDuplicatedNoNeed) = BatchDuplicatedNoNeed(adapt(to, x.val), adapt(to, x.dval))
 
+"""
+    congruent(a::T, b::T)::T
+
+Defines values to be congruent, e.g. structurally equivalent.
+"""
+function congruent end
+
+congruent(a::T, b::T) where T<:Number = true
+congruent(a::T, b::T) where T<:AbstractArray = length(a) == length(b)
```

Comment:
This is not sufficient, say for an offset array they should have same offset likely

Reply:
I think `axes(a) == axes(b)` should suffice.